### ISSUE TITLE
How to launch my own FWBuild server?

### ISSUE THREAD

**Issue by Harvie** [cesanta/mos-tool#40]
`mos help --full` mentions this parameter:
`      --server string                       FWBuild server (default "https://mongoose.cloud")`

How can i install such service on my server, so we don't have to get docker running on all of our developer laptops? Default mongoose.cloud server has some time limit, which is too small for us. even for relatively small test projects. So i would like to have my own server without time limit.

**Comment by rojer** [comment]
you need to run fwbuild-manager. the source is here, there's also a dockker-compose file that should give you an idea: https://github.com/mongoose-os/mos/tree/master/fwbuild

there is currently no documentation, if you can contribute a README that would be great.

**Comment by rojer** [comment]
also, the new location for the mos repo is https://github.com/mongoose-os/mos, please move all activity there.

**Comment by amizer12** [comment]
@Harvie Did you ever end up completing this ? I`m at the same stage when i need to setup a build server remotely ...

**Comment by Harvie** [comment]
> @Harvie Did you ever end up completing this ?


Not really, but i still think it would be useful as some of my coworkers were still not able to get it running localy and i dont want to spend time fixing everyone's laptop.

**Comment by amizer12** [comment]
@Harvie Thanks - wil post here if i will be able to get anywhere - greetings form a fellow Bill Hicks fan :)

**Comment by amizer12** [comment]
@Harvie 
i`ve added instructions how to set this up to main Readme file - hope that helps - [https://github.com/mongoose-os/mos#setup-a-private-build-server--ubuntu-](url)

**Comment by Harvie** [comment]
@amizer12 cool! seems simple. thank you.
i wonder if there is way to increase the file size limit. removing the deps seems bit problematic to me, since i sometimes have local modifications made in the deps during the development and debuging. i also have local deps without online repository.

**Comment by amizer12** [comment]
@Harvie editing /mos/fwbuild/manager/fwbuild_manager.go will change the file size limit - see line :
`payloadLimit      = flag.Int64("payload-size-limit", 5*1024*1024, "Max upload size")`

**Comment by rojer** [comment]
local deps - i.e. libs with `location: /path` - should be working already.
not sending clean deps to server is a long time TODO, for now we just ship everything.
